python/src/aiconfig/editor/client/src/components/SettingsPropertyRenderer.tsx

```diff
@@ -124,6 +125,33 @@ export default function SettingsPropertyRenderer({
     ]);
   }, [property.items, removeItemFromList, setAndPropagateValue]);
 
+ 
+
+  useEffect(() => {
```

Comment:
We don't need an effect for this. This is just setting the initial state for itemControls so we should instead just initialize it inline above using the init function for useState:

```
const [itemControls, setItemControls] = useState<JSX.Element[]>(() => {
  if (!Array.isArray(initialValue)) {
    return [];
  }

  return initialValue.map((arrayItem,index) => {
    const key = uniqueId();
    return (
      <Group key={key}>
        <SettingsPropertyRenderer
          propertyName=""
          property={property.items}
          initialValue={arrayItem}
          setValue={newItem => {
            itemValues.current.set(key, newItem);
            setAndPropagateValue(Array.from(itemValues.current.values()));
          }}
       />
       <ActionIcon onClick={() => removeItemFromList(key)}>
         <IconTrash size={16} />
       </ActionIcon>
     </Group>
    );
  });
});
```

Reply:
updated as suggested. Is there anything bad that would happen if we were to continue to use `useEffect()`?

Reply:
Nothing critically bad, it's just the scenario I told you about --> doing it in a useEffect would first render with default [] in state, then hit the effect and then re-render again with new state from the effect. It's an extra full render cycle. This way the state is set initially before anything is ever added to the DOM and no re-render is needed